### PR TITLE
Vue: Add Customization for Vue Using Slots

### PR DESCRIPTION
**Pull request by ErikCH**
*Issue #, if available:*

*Description of changes:*

This PR adds functionality so users can overwrite the sign up fields, and include their own custom attributes and functionality. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/vue/src/composables/useAuth.ts [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+    const facadeValues = getServiceFacade({ send, state: state.value });
+    for (const key of Object.keys(facade)) {
+      //@ts-ignore
+      useAuthenticatorValue[key] = facadeValues[key];
+    }
+    useAuthenticatorValue.send = send;
+    useAuthenticatorValue.state = state;
+  });
+  return useAuthenticatorValue;
 };
```
[the remark this answers is not in the document]
Comment (ErikCH) [reply packages/vue/src/composables/useAuth.ts]: After further though, I'm going to surround this in a `onMount` hook, just in case. It should be fine.

File: examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    <template v-slot:sign-up-fields>
+      <base-text-field
+        id="prefered_username"
+        label="Preferred Username"
+        name="preferred_username"
+        placeholder="Preferred Username"
+        type="text"
+        :required="false"
+      ></base-text-field>
+      <sign-up-form-fields />
```
[the remark this answers is not in the document]
Comment (ErikCH) [reply examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue]: I'm good with either way, I thought at first it was too wordy, but maybe doesn't matter...

File: examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import Amplify from 'aws-amplify';
+import { toRefs } from 'vue';
+import {
+  Authenticator,
+  SignUpFormFields,
+  // Vue Composable to get access to validation errors
+  useAuthenticator,
+  // Amplify UI Primitives to simplify the custom fields
+  BaseTextField,
+  BaseCheckBox,
```
[the remark this answers is not in the document]
Comment (ericclemmons) [reply examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue:10]: Since this is our documented public API, I think we should figure out our naming & guidance for the future as well.

https://v3.vuejs.org/style-guide/#base-component-names-strongly-recommended are great recommendations for apps, but I do wonder what a 3rd party library primitives should be?

For comparison, Vuetify uses `v-alert` – https://vuetifyjs.com/en/components/alerts/

To avoid conflicts, does this seem reasonable?

- `base-checkbox` is a *customer's* base checkbox implementation.
- `v-alert` is from Vuetify (hence the `v`)
- `amplify-checkbox` is _ours_.

File: examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+} from '@aws-amplify/ui-vue';
+import '@aws-amplify/ui-vue/styles.css';
+import awsExports from '@environments/auth-with-email-and-custom-attributes/src/aws-exports';
+
+Amplify.configure(awsExports);
+
+const { validationErrors } = toRefs(useAuthenticator());
```
[the remark this answers is not in the document]
Comment (ericclemmons) [reply examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue:18]: Ooh my mistake!  It's the _destructing_ that's making it a reactive thing.  So this _wouldn't_ require `toRefs`:

```js
const authenticator = useAuthenticator();

...
if (authenticator.validationErrors) ...
```

Nevermind, I withdraw this!

**Review by ericclemmons — COMMENTED** [review]
This is looking really good!  The only thing I want to determine today is what our Vue component prefix will be to avoid conflicts with (1) customer's own abstractions & (2) other popular 3rd party libraries.

That'll make the documented API stable & consistent for the foreseeable future 🙏

File: examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (ErikCH) [reply examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue:10]: OK, I can do that. I can add `amplify` to it. I will leave the internal ones with `base` though, unless they get exposed to the customer.

[the remark this answers is not in the document]
Comment (ErikCH) [reply examples/vue/src/pages/ui/components/authenticator/custom-sign-up-fields/index.vue]: Pushed up :) @ericclemmons

**Review by ericclemmons — APPROVED** [review]
Looking much better!
